src/functTest/groovy/com/bmuschko/gradle/docker/tasks/image/DockerRemoveImageFunctionalTest.groovy

```diff
@@ -0,0 +1,87 @@
+package com.bmuschko.gradle.docker.tasks.image
+
+import com.bmuschko.gradle.docker.AbstractFunctionalTest
+import com.bmuschko.gradle.docker.TestPrecondition
+import org.gradle.testkit.runner.BuildResult
+import spock.lang.Requires
+
+import static org.gradle.testkit.runner.TaskOutcome.SKIPPED
+import static org.gradle.testkit.runner.TaskOutcome.SUCCESS
+
+@Requires({ TestPrecondition.DOCKER_SERVER_INFO_URL_REACHABLE })
+class DockerRemoveImageFunctionalTest extends AbstractFunctionalTest {
+
+    def "can remove image"() {
+        buildFile << """
+            import com.bmuschko.gradle.docker.tasks.image.Dockerfile
+            import com.bmuschko.gradle.docker.tasks.image.DockerBuildImage
+            import com.bmuschko.gradle.docker.tasks.image.DockerRemoveImage
+
+            task dockerfile(type: Dockerfile) {
+                from 'ubuntu:12.04'
+            }
+
+            task buildImage(type: DockerBuildImage) {
+                dependsOn dockerfile
+                inputDir = file("build/docker")
+            }
+            
+            task removeImage(type: DockerRemoveImage) {
+                dependsOn buildImage
+                targetImageId { buildImage.getImageId() }
+            }
+        """
+
+        when:
+        BuildResult result = build('removeImage')
+
+        then:
+        result.output.contains("Removing image with ID")
+    }
+
+    def "can remove image tagged in multiple repositories"() {
+        buildFile << """
+            import com.bmuschko.gradle.docker.tasks.image.Dockerfile
+            import com.bmuschko.gradle.docker.tasks.image.DockerBuildImage
+            import com.bmuschko.gradle.docker.tasks.image.DockerRemoveImage
+            import com.bmuschko.gradle.docker.tasks.image.DockerTagImage
+
+            task dockerfile(type: Dockerfile) {
+                from 'ubuntu:12.04'
+            }
+
+            task buildImage(type: DockerBuildImage) {
+                dependsOn dockerfile
+                inputDir = file("build/docker")
+            }
+            
+            task tagImage(type: DockerTagImage) {
+                dependsOn buildImage
+                repository = "repository"
+                tag = "tag2"
+                targetImageId { buildImage.getImageId() }
+            }
+            
+            task tagImageSecondTime(type: DockerTagImage) {
+                dependsOn tagImage
+                repository = "repository"
+                tag = "tag2"
+                targetImageId { buildImage.getImageId() }
+            }
+            
+            task removeImage(type: DockerRemoveImage) {
+                dependsOn tagImageSecondTime
+                force = true
+                targetImageId { buildImage.getImageId() }
+            }
+        """
+
+        when:
+        BuildResult result = build('removeImage')
+
+        then:
+        result.output.contains("Removing image with ID")
```

Comment:
Are you sure that this test actually works? In `DockerRemoveImage` logging of _"Removing image with ID"_ is done __before__ executing remove command in docker engine.

It would be better to execute another task of type `DockerListImages` and assert that target image is not in the result.